charmtools/charms.py

```diff
@@ -500,6 +502,30 @@ def schema_type(self, **kw):
         )
 
 
+def validate_min_juju_version(charm, linter):
+    """Validate min-juju-version in charm metadata.
+
+    Must match the regex and be 2.0.0 or greater.
+
+    :param charm: dict of charm metadata parsed from metadata.yaml
+    :param linter: :class:`CharmLinter` object to which info/warning/error
+        messages will be written
+
+    """
+    if 'min-juju-version' not in charm:
+        return
+
+    pattern = r'^(\d{1,9})\.(\d{1,9})(\.|-(\w+))(\d{1,9})(\.\d{1,9})?$'
```

Comment:
Well we could do `\d[2-9]{1,9}`, but my point with the original version was to give them a more specific error msg if the format was okay, but the version wasn't high enough. If you think that's not important I'll change it.